github/table_github_repository_content.go

```diff
@@ -0,0 +1,193 @@
+package github
+
+import (
+	"context"
+
+	"github.com/shurcooL/githubv4"
+	"github.com/turbot/steampipe-plugin-github/github/models"
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableGitHubRepositoryContent() *plugin.Table {
+	return &plugin.Table{
+		Name:        "github_repository_content",
+		Description: "List the content in a repository (list directory, or get file content",
+		List: &plugin.ListConfig{
+			Hydrate:           tableGitHubRepositoryContentList,
+			ShouldIgnoreError: isNotFoundError([]string{"404"}),
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "repository_full_name", Require: plugin.Required},
+				{Name: "repository_content_path", Require: plugin.Optional, CacheMatch: "exact"},
+			},
+		},
+		Columns: []*plugin.Column{
+			{Name: "repository_full_name", Description: "The full name of the repository (login/repo-name).", Type: proto.ColumnType_STRING, Transform: transform.FromQual("repository_full_name")},
+			{Name: "type", Description: "The file type (directory or file).", Type: proto.ColumnType_STRING},
+			{Name: "name", Description: "The file name.", Type: proto.ColumnType_STRING},
+			{Name: "oid", Description: "The Git object ID.", Type: proto.ColumnType_STRING},
+			{Name: "abbreviated_oid", Description: "An abbreviated version of the Git object ID.", Type: proto.ColumnType_STRING},
+			{Name: "repository_content_path", Description: "The requested path in repository search.", Type: proto.ColumnType_STRING, Transform: transform.FromQual("repository_content_path")},
+			{Name: "path", Description: "The path of the file.", Type: proto.ColumnType_STRING},
+			{Name: "path_raw", Description: "A Base64-encoded representation of the file's path.", Type: proto.ColumnType_STRING},
+			{Name: "mode", Description: "The mode of the file.", Type: proto.ColumnType_INT},
+			{Name: "size", Description: "The size of the file (in KB).", Type: proto.ColumnType_INT},
+			{Name: "line_count", Description: "The number of lines available in the file.", Type: proto.ColumnType_INT},
+			{Name: "content", Description: "The decoded file content (if the element is a file).", Type: proto.ColumnType_STRING},
+			{Name: "is_generated", Description: "Whether or not this tree entry is generated.", Type: proto.ColumnType_BOOL},
+			{Name: "is_binary", Description: "Indicates whether the Blob is binary or text.", Type: proto.ColumnType_BOOL},
+			{Name: "sha", Description: "The sha of the file.", Type: proto.ColumnType_STRING, Transform: transform.FromField("CommitUrl").Transform(lastPathElement)},
```

Comment:
I think this is the same value in `oid` - at least in the couple of tests I did.